src/core/execute.c

```diff
@@ -1213,13 +1213,15 @@ static int setup_pam(
 
                 if (getttyname_malloc(STDIN_FILENO, &q) >= 0)
                         tty = strjoina("/dev/", q);
+                else
+                        /* If everything else failed then let's just use value "systemd". This will cause that session
+                         * isn't going to be marked as "background" and user manager will be started. */
+                        tty = "systemd";
         }
```

Comment:
I don't follow here? PAMName= without a tty pretty much *is* background stuff? so why mark it specially? What do you have in mind that sets PAMName= without tty here that you don't want to affect?

Reply:
(I am a bit conservative about adding more special somethings to the PAM tty field that encodes stuff that clearly is not about actual ttys)